openfe/protocols/openmm_utils/system_creation.py

```diff
@@ -0,0 +1,236 @@
+    thermo_settings : ThermoSettings
+      Thermodynamic settings, including everything necessary to
+      create a barostat.
+    system_settings : SystemSettings
+      System settings including all necessary information for
+      the nonbonded methods.
+    cache : pathlib.Path
+      Path to openff force field cache.
+    has_solvent : bool
+      Wether or not the target system has solvent (and by extension
+      might require a barostat).
+
+    Returns
+    -------
+    system_generator : openmmforcefields.generator.SystemGenerator
+      System Generator to use for this Protocol.
+
+    TODO
+    ----
+    * Investigate how RF can be passed to non-periodic kwargs.
+    """
+    # get the right constraint
+    constraints = {
+        'hbonds': app.HBonds,
+        'none': None,
+        'allbonds': app.AllBonds,
+        'hangles': app.HAngles
+        # vvv can be None so string it
+    }[str(forcefield_settings.constraints).lower()]
+
+    # create forcefield_kwargs entry
+    forcefield_kwargs = {
+        'constraints': constraints,
+        'rigidWater': forcefield_settings.rigid_water,
+        'removeCMMotion': forcefield_settings.remove_com,
+        'hydrogenMass': forcefield_settings.hydrogen_mass * omm_unit.amu,
+    }
+
+    # get the right nonbonded method
+    nonbonded_method = {
+        'pme': app.PME,
+        'nocutoff': app.NoCutoff,
+        'cutoffnonperiodic': app.CutoffNonPeriodic,
+        'cutoffperiodic': app.CutoffPeriodic,
+        'ewald': app.Ewald
+    }[system_settings.nonbonded_method.lower()]
+
+    nonbonded_cutoff = to_openmm(
+        system_settings.nonbonded_cutoff
+    )
+
+    # create the periodic_kwarg entry
+    periodic_kwargs = {
+        'nonbondedMethod': nonbonded_method,
+        'nonbondedCutoff': nonbonded_cutoff,
+    }
+
+    # Currently the else is a dead branch, we will want to investigate the
+    # possibility of using CutoffNonPeriodic at some point though (for RF)
+    if nonbonded_method is not app.CutoffNonPeriodic:
+        nonperiodic_kwargs = {
+                'nonbondedMethod': app.NoCutoff,
+        }
+    else:  # pragma: no-cover
+        nonperiodic_kwargs = periodic_kwargs
+
+    system_generator = SystemGenerator(
+        forcefields=forcefield_settings.forcefields,
+        small_molecule_forcefield=forcefield_settings.small_molecule_forcefield,
+        forcefield_kwargs=forcefield_kwargs,
+        nonperiodic_forcefield_kwargs=nonperiodic_kwargs,
+        periodic_forcefield_kwargs=periodic_kwargs,
+        cache=str(cache),
+    )
+
+    # Add a barostat if necessary
+    # TODO: move this to its own place where we can then handle membranes
+    # Note: this behaviour was broken pre 0.11.2 of openmmff
+    if has_solvent:
+        barostat = MonteCarloBarostat(
+            ensure_quantity(thermo_settings.pressure, 'openmm'),
+            ensure_quantity(thermo_settings.temperature, 'openmm'),
+        )
+        system_generator.barostat = barostat
```

Comment:
There was, but it's been so long since writing code to review that I don't remember anymore. I'm happy moving it back to SystemGenerator, but if there's an issue we'll have to find it out later.